compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -443,9 +450,16 @@ pub(crate) fn inline_asm_call(
     alignstack: bool,
     dia: LlvmAsmDialect,
     line_spans: &[Span],
+    unwind: bool,
```

Comment:
This `bool` is redundant with the `Option` just below it.

Reply:
I don’t think that’s enough because it’s possible we might want to emit an unwinding ‘call asm’ instruction (in that case ‘unwind’ is true and the option is ’None’.

Reply:
For reference when there is nothing to cleanup on unwind in the current function, `cleanup` will be `None` and a regular `call` instruction will be used. `invoke` is only necessary when you want to actually run code on unwinding.

Reply:
Yes. But the inline assembly still has to be marked as “unwind” to be actually sound because eg. the llvm inlining pass needs to know that the ‘call asm’ might unwind (inlining  a ‘call asm unwind’ into an unwinding function might turn the ‘call asm unwind’ into an ‘invoke asm unwind’. The inliner isn’t performing that transformation when the ‘call asm’ isn’t marked as ‘unwind’).